### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by KonanAl**
resquet 1 : dataType for loading and saving functions

### CODE REVIEW

**Review by deaspo — CHANGES_REQUESTED** [review]
- Only submit changes that have been tested
- Avoid usage of hardcorded values
- Leave untouched functions that you haven't tested
- Test the library and address all warnings, avoid duplicating implementations

File: ImageInLib/ImageInLib/src/noise_generator.h [hunk 1]
```diff
@@ -35,14 +35,14 @@ extern "C" {
...
-	void addMultiplicativeNoise(dataType **imageDataPtr, size_t imageHeight, size_t imageLength, size_t imageWidth, float variance, dataType fgMin, dataType bgMax);
+	void addMultiplicativeNoise(dataType **imageDataPtr, size_t imageHeight, size_t imageWidth, dataType variance);
 	/*
 	* Structural noise generate adds noise to imageDataPtr
 	* imageDataPtr contains points that we will add noise to
 	* imageHeight represent the Z coordinate, imageWidth is the X*Y coordinate - 2D representation of 3D
 	* Uses a sinusoid function to generate regular noise
 	*/
-	void addStructuralNoise(dataType **imageDataPtr, size_t imageHeight, size_t imageLength, size_t imageWidth, dataType fgMin, dataType bgMax);
+	void addStructuralNoise(dataType **imageDataPtr, int imageHeight, int imageWidth);
```
[the remark this answers is not in the document]
Comment (KonanAl) [reply ImageInLib/ImageInLib/src/noise_generator.h:45]: I really didn't test functions for adding structural noises (In my second version of the library), so I supposed that they are in they original version. But I see and understand the issues your pointed, I will fix them, thanks.

File: ImageInLib/ImageInLib/include/common_functions.h [hunk 2]
```diff
@@ -53,6 +59,12 @@ extern "C" {
 		size_t num; // shape number
 	} Shapes;
 	//==============================================================================
+	typedef struct {
+		float salt_pepper_density, multiplicative_variance;
+		int additive_value;
+		dataType fgMin, bgMax;
+	} NoiseParameters;
```
[the remark this answers is not in the document]
Comment (KonanAl) [reply ImageInLib/ImageInLib/include/common_functions.h:66]: It was not defined in my version of noiseGeneration.h. But, I move it there, now

**Comment by deaspo** [comment]
Synchronize your master and updates branch before making a pull request to the imageInLib master branch.

**Pull request closed without merging**